crates/starknet_api/src/crypto/utils.rs

```diff
@@ -78,7 +78,7 @@ pub fn verify_message_hash_signature(
 }
 
 // Collect elements for applying hash chain.
-pub(crate) struct HashChain {
+pub struct HashChain {
```

Comment:
Don't make changes outside the blockifier folder. Please revert this and find another solution

Reply:
Changed to use `BlockHash(Poseidon::hash_array(...))`